spec/lib/appsignal/integrations/ownership_spec.rb

```diff
@@ -0,0 +1,149 @@
+          # the code within the `owner` block, as documented in `ownership`'s
+          # README:
+          # https://github.com/ankane/ownership/blob/b277ef821654d0e73d2e6c8df4f636932b7a90fa/README.md#custom-integrations
+          block.call
+        end
+
+        expect(override).to receive(:call).with("owner", kind_of(Proc)).and_call_original
+
+        Ownership.around_change = override
+
+        transaction = Appsignal::Transaction.create("namespace")
+
+        block = proc {}
+        expect(block).to receive(:call)
+
+        owner("owner", &block)
+
+        expect(transaction.namespace).to eq("owner")
+      end
+    end
+
+    context "when an error is reported in a transaction" do
+      it "sets the namespace of the transaction to the owner where the error was raised" do
+        transaction = Appsignal::Transaction.create("namespace")
+
+        begin
+          owner("error") do
+            raise "error"
+          end
+        rescue StandardError => error
+          # This owner should be overriden on the namespace by the error owner.
+          owner("rescue") do
+            nil
+          end
+          expect(transaction.namespace).to eq("rescue")
+
+          transaction.add_error(error)
+          transaction.complete
+          expect(transaction.namespace).to eq("error")
+        end
+      end
+    end
+
+    context "when several errors are reported in a transaction" do
+      it "sets the namespace of the transaction to the owner where the last error was raised" do
+        transaction = Appsignal::Transaction.create("namespace")
+
+        begin
+          owner("first") do
+            raise "first error"
+          end
+        rescue StandardError => first_error
+          # This owner should be overriden on the namespace by the error owner.
+          owner("first_rescue") do
+            nil
+          end
+
+          transaction.add_error(first_error)
+        end
+
+        begin
+          owner("second") do
+            raise "second error"
+          end
+        rescue StandardError => second_error
+          # This owner should be overriden on the namespace by the error owner.
+          owner("second_rescue") do
+            nil
+          end
+
+          transaction.add_error(second_error)
+        end
+
+        # This owner should be overriden on the namespace by the error owner.
+        owner("after_rescues") do
+          nil
+        end
+
+        # We keep track of the transactions that are completed using the
+        # `before_complete` hook -- otherwise, we do not have access to the
+        # duplicate transaction.
+        seen_transactions = {}
+        Appsignal::Transaction.before_complete do |t, error|
+          seen_transactions[error.message] = t
+        end
+
+        transaction.complete
+
+        expect(seen_transactions["first error"].namespace).to eq("first")
+        expect(seen_transactions["second error"].namespace).to eq("second")
```

Comment:
Does this need to be caught via `before_complete` or could you use the [`created_transactions` helper](https://github.com/appsignal/appsignal-ruby/blob/e2d186c9e79a5d03d3692c6ed84ca085e1383031/spec/support/helpers/transaction_helpers.rb#L52-L58)?

Reply:
Yes, that would be preferable, thank you! I had a vague memory that a helper like that existed, but I couldn't find it.